test/serializers/attributes_test.rb

```diff
@@ -18,11 +18,31 @@ def test_attributes_definition
           @profile_serializer.class._attributes)
       end
 
+      def test_attributes
+        assert_equal({ name: 'Name 1', description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
       def test_attributes_with_fields_option
         assert_equal({ name: 'Name 1' },
           @profile_serializer.attributes(fields: [:name]))
       end
 
+      def test_attributes_with_fields_method
+        @profile_serializer.define_singleton_method(:fields) do |keys|
+          keys - [:name]
+        end
+        assert_equal({ description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
+      def test_attributes_with_fields_option_and_method
+        @profile_serializer.define_singleton_method(:fields) do |keys|
```

Comment:
Because you wrote the method fields, you should test how that plays with various seriolizers. This test is testing the overridden method, which wouldn't touch the new code you wrote.